### PR TITLE
New RN summary

### PR DESCRIPTION
**Pull request by dantavori**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23756

## Description
Added the new release notes summary generator (in packs format) to the build process. Currently in `Build Content Descriptor` step, right after `release_notes.py` execution.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/scripts/infrastructure_tests/release_notes_generator_test.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        Given
+        - A release note file with an Integration and a Script:
+            - FakePack4_Script1
+            - FakePack4_Integration1
+
+        When
+        - Formatting a release notes file.
+
+        Then
+        - Ensure both integration appear in the formatted string
```
Comment (Itay4) [inline Tests/scripts/infrastructure_tests/release_notes_generator_test.py:64]: this seems wrong

File: release_notes_generator.py [hunk 2]
```diff
@@ -0,0 +1,408 @@
+from __future__ import print_function
```
Comment (Itay4) [inline release_notes_generator.py:1]: should this file be in content root?

File: release_notes_generator.py [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+                    'setting the upstream, run "git fetch upstream", and then run "git merge upstream/master". Doing '
+                    'these steps will merge your branch with content master as a base.')
+        sys.exit(1)
+
+    pack_paths = [os.path.dirname(file_path) for file_path in diff_result.split('\n')
+                  if file_path.startswith(PACKS_DIR)]
+    return pack_paths
+
+
+def get_new_entity_record(entity_path: str) -> (str, str):
```
Comment (Itay4) [inline release_notes_generator.py:63]:
```suggestion
def get_new_entity_record(entity_path: str) -> Tuple[str, str]:
```

File: release_notes_generator.py [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+
+def get_pack_entities(pack_path):
+    print(f'Processing "{pack_path}" files:')
+    pack_entities = sum([
+        glob.glob(f'{pack_path}/*/*.json'),
+        glob.glob(f'{pack_path}/*/*.yml'),
+        glob.glob(f'{pack_path}/*/*/*.yml')], [])
+    pack_entities.sort()
+
+    entities_data = {}
```
Comment (Itay4) [inline release_notes_generator.py:125]:
```suggestion
    entities_data: Dict = {}
```

File: release_notes_generator.py (hunk 2 above)
Comment (yaakovi) [reply release_notes_generator.py:1]: the old release notes generator was there, do you suggest something else?
maybe `Utils`?

Comment (Itay4) [reply release_notes_generator.py:1]: yea

**Pull request merged**